From: Jan Engelhardt <jengelh@inai.de>
Date: 2018-06-04 09:47:12.535168981 +0200
References: http://bugzilla.suse.com/show_bug.cgi?id=1095425

Because of bug #1095425, IS_UTF8=0 is wanted. Since that changes the C++ABI of
class UnicodeString, the library would become incompatible to a default build
that 3rd party programs might expect, and so we need to change the SO
identifier to something unique.

---
 source/common/unicode/platform.h |    2 +-
 source/icudefs.mk.in             |    2 +-
 2 files changed, 2 insertions(+), 2 deletions(-)

Index: icu/source/common/unicode/platform.h
===================================================================
--- icu.orig/source/common/unicode/platform.h
+++ icu/source/common/unicode/platform.h
@@ -632,7 +632,7 @@ namespace std {
 #ifdef U_CHARSET_IS_UTF8
     /* Use the predefined value. */
 #elif U_PLATFORM_IS_LINUX_BASED || U_PLATFORM_IS_DARWIN_BASED
-#   define U_CHARSET_IS_UTF8 1
+#   define U_CHARSET_IS_UTF8 0
 #else
 #   define U_CHARSET_IS_UTF8 0
 #endif
Index: icu/source/icudefs.mk.in
===================================================================
--- icu.orig/source/icudefs.mk.in
+++ icu/source/icudefs.mk.in
@@ -41,7 +41,7 @@ PACKAGE_ICU_URL = "http://icu-project.or
 PACKAGE = @PACKAGE@
 VERSION = @VERSION@
 UNICODE_VERSION = @UNICODE_VERSION@
-SO_TARGET_VERSION = @LIB_VERSION@
+SO_TARGET_VERSION = suse@LIB_VERSION@
 SO_TARGET_VERSION_MAJOR = @LIB_VERSION_MAJOR@
 
 # The ICU data external name is usually icudata; the entry point name is
